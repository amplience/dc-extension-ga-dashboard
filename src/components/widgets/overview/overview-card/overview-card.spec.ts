import { render } from '@testing-library/svelte';
import OverviewCard from './overview-card.svelte';

describe('OverviewCard', () => {
  it('should render the OverviewCard component', () => {
    const { container } = render(OverviewCard, {
      title: 'test title',
      icon: 'test-icon',
      value: 1,
      bodyText: 'some body text',
      color: '#333',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the OverviewCard component with footerText', () => {
    const { container } = render(OverviewCard, {
      title: 'test title',
      icon: 'test-icon',
      value: 1,
      bodyText: 'some body text',
      footerText: 'some footer text',
      color: '#333',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the OverviewCard component when no value is set', () => {
    const { container } = render(OverviewCard, {
      title: 'test title',
      icon: 'test-icon',
      value: null,
      bodyText: 'some body text',
      color: '#333',
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
