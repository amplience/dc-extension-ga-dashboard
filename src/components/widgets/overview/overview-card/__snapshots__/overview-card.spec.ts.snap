// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewCard should render the OverviewCard component 1`] = `
<div>
  <article
    class="svelte-14k6rag"
    style="--overview-card-color: #333;"
  >
    <div
      class="overview-card-top svelte-14k6rag"
    >
      <div
        class="overview-card-heading svelte-14k6rag"
      >
        <div
          class="overview-card-title svelte-14k6rag"
        >
          <div
            class=" svelte-zpbvvv"
            style="--icon-width: 24px; --icon-height: 24px;"
          >
            test-icon
          </div>
           
          <h2
            class="svelte-14k6rag"
          >
            test title
          </h2>
        </div>
         
        <div
          class="overview-card-value svelte-14k6rag"
        >
          1
        </div>
      </div>
       
      <div
        class="overview-card-content svelte-14k6rag"
      />
    </div>
     
    <div
      class="overview-card-footer svelte-14k6rag"
    />
  </article>
</div>
`;

exports[`OverviewCard should render the OverviewCard component when no value is set 1`] = `
<div>
  <article
    class="svelte-14k6rag"
    style="--overview-card-color: #333;"
  >
    <div
      class="overview-card-top svelte-14k6rag"
    >
      <div
        class="overview-card-heading svelte-14k6rag"
      >
        <div
          class="overview-card-title svelte-14k6rag"
        >
          <div
            class=" svelte-zpbvvv"
            style="--icon-width: 24px; --icon-height: 24px;"
          >
            test-icon
          </div>
           
          <h2
            class="svelte-14k6rag"
          >
            test title
          </h2>
        </div>
         
        <div
          class="overview-card-value svelte-14k6rag"
        >
          -
        </div>
      </div>
       
      <div
        class="overview-card-content svelte-14k6rag"
      />
    </div>
     
    <div
      class="overview-card-footer svelte-14k6rag"
    />
  </article>
</div>
`;

exports[`OverviewCard should render the OverviewCard component with footerText 1`] = `
<div>
  <article
    class="svelte-14k6rag"
    style="--overview-card-color: #333;"
  >
    <div
      class="overview-card-top svelte-14k6rag"
    >
      <div
        class="overview-card-heading svelte-14k6rag"
      >
        <div
          class="overview-card-title svelte-14k6rag"
        >
          <div
            class=" svelte-zpbvvv"
            style="--icon-width: 24px; --icon-height: 24px;"
          >
            test-icon
          </div>
           
          <h2
            class="svelte-14k6rag"
          >
            test title
          </h2>
        </div>
         
        <div
          class="overview-card-value svelte-14k6rag"
        >
          1
        </div>
      </div>
       
      <div
        class="overview-card-content svelte-14k6rag"
      />
    </div>
     
    <div
      class="overview-card-footer svelte-14k6rag"
    />
  </article>
</div>
`;
